script/cadence/contracts/FlowColdStorageProxy.cdc

```diff
@@ -124,10 +150,143 @@ pub contract FlowColdStorageProxy {
             emit Transferred(from: self.owner!.address, to: receiver, amount: amount)
         }
 
+        pub fun delegateVaultTokens(amount: UFix64, nonce: Int64, sig: [UInt8]) {
+            // Ensure that the nonce follows on from the previous meta
+            // transaction.
+            if nonce != (self.lastNonce + 1) {
+                panic("Invalid meta transaction nonce")
+            }
+
+            // Construct the message to sign.
+            var data: [UInt8] = "delegate".toBytes()
+            data = data.concat(amount.toBigEndianBytes())
+            data = data.concat(nonce.toBigEndianBytes())
+
+            // Verify the signature.
+            let key = PublicKey(
+                publicKey: self.publicKey,
+                signatureAlgorithm: SignatureAlgorithm.ECDSA_secp256k1
+            )
+            if !key.verify(signature: sig, signedData: data, domainSeparationTag: "FLOW-V0.0-user", hashAlgorithm: HashAlgorithm.SHA3_256) {
+                panic("Invalid meta transaction signature")
+            }	
```

Comment:
A lot of this code is boilerplate and is the same between the different methods. Would it be possible to split some of it out into utility methods to avoid boilerplate?